### PR TITLE
Updated the Workflow articles to Symfony 4

### PR DESCRIPTION
**Pull request by javiereguiluz**
The new file config paths are pending of this: https://github.com/symfony/recipes/pull/256

### CODE REVIEW

**Comment by xabbuh** [comment]
symfony/recipes#256 has been merged

File: workflow/state-machines.rst [hunk 1]
```diff
@@ -190,8 +189,23 @@ Below is the configuration for the pull request state machine.
...
+    // ...
+    use Symfony\Component\Workflow\Registry;
+
+    class SomeService
+    {
+        private $stateMachine;
+
+        public function __constructor(Registry $workflows)
+        {
+            $this->stateMachine = $workflows->get('pull_request');
```
Comment (xabbuh) [inline workflow/state-machines.rst]: I would not do that inside the constructor. You'll never know when your service is initialized and if the registry is ready at this point which could lead to undesired side effects.

File: workflow/usage.rst [hunk 2]
```diff
@@ -152,28 +159,43 @@ like this:
...
+    use Symfony\Component\Workflow\Registry;
 
-    $workflow = $this->container->get('workflow.blog_publishing');
-    $workflow->can($post, 'publish'); // False
-    $workflow->can($post, 'to_review'); // True
+    class BlogController
+    {
+        public function edit(Registry $workflows)
+        {
+            $post = new \App\Entity\BlogPost();
```
Comment (xabbuh) [inline workflow/usage.rst]: I would add a use statement for the class instead of using the FQCN here.

File: workflow/usage.rst [hunk 3]
```diff
@@ -152,28 +159,43 @@ like this:
...
+            $workflow->can($post, 'to_review'); // True
+
+            // Update the currentState on the post
+            try {
+                $workflow->apply($post, 'to_review');
+            } catch (LogicException $e) {
+                // ...
+            }
+
+            // See all the available transition for the post in the current state
```
Comment (xabbuh) [inline workflow/usage.rst]: transitions

File: workflow/usage.rst [hunk 4]
```diff
@@ -250,7 +272,8 @@ order:
     * ``workflow.[workflow name].announce``
     * ``workflow.[workflow name].announce.[transition name]``
 
-Here is an example how to enable logging for every time a the "blog_publishing" workflow leaves a place::
+Here is an example how to enable logging for every time the ``blog_publishing``
```
Comment (xabbuh) [inline workflow/usage.rst]: [...] an example of [...]

**Review by xabbuh — APPROVED** [review]

**Review by weaverryan — APPROVED** [review]

**Pull request merged**